CMakeLists.txt

```
@@ -18,25 +18,45 @@ endif ()
 # in the welcome screen, and hiding of some development menu commands, but
 # still link to the alpha manual online.
 
+# Just a couple of convenience variables
+set( topdir "${CMAKE_SOURCE_DIR}" )
+set( libsrc "${topdir}/lib-src" )
+
 # Set this value to 0 for alpha, 1 for beta, 2 for release builds
 set( AUDACITY_BUILD_LEVEL 0 )
 
 # The Audacity version
-# Increment as appropriate after release of a new version, and set back
-# AUDACITY_BUILD_LEVEL to 0
-set( AUDACITY_VERSION 3 )
+set( AUDACITY_VERSION 0 )
 set( AUDACITY_RELEASE 0 )
-set( AUDACITY_REVISION 4 )
-set( AUDACITY_MODLEVEL 0 )
-
-if( AUDACITY_BUILD_LEVEL EQUAL 0 )
-   set( AUDACITY_SUFFIX "-alpha" )
-elseif(  AUDACITY_BUILD_LEVEL EQUAL 1 )
-   set( AUDACITY_SUFFIX "-beta" )
-else()
-   set( AUDACITY_SUFFIX "" )
+set( AUDACITY_REVISION 0 )
+find_package( Git QUIET )
+if( GIT_FOUND )
+    execute_process(
+        COMMAND
+            ${GIT_EXECUTABLE} describe --abbrev=0
+        WORKING_DIRECTORY
+            ${topdir}
```

```suggestion
            ${CMAKE_SOURCE_DIR}
```
This `${topdir}` is overcomplicated. It does not make the code easier to read IMO; it only makes me wonder why the `${topdir}` variable exists and look up where it is set to understand what it does.

---

`libsrc` isn't used very much either, but I think that should be a separate cleanup PR.

---

I agree. Here I am only asking not to use it in new code.

---

Fair. I'll put it back where it was then

---

You put the old code back where it was but the new code is still using `${topdir}`

---

Yes, I fixed only one instance of it. one sec